tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -2979,4 +2979,311 @@ public function testEnsurePrimaryKeyBeingReadFromTableWhenLoadingBelongsToManyRe
         ];
         $this->assertEquals($expected, $result->toArray());
     }
+
+    /**
+     * Test all scenarios
+     *
+     * @return void
+     */
+    public function testHasTranslations()
+    {
+        // Translate Behavior not attached
+        $table = TableRegistry::get('Articles');
+
+        $marshall = new Marshaller($table);
+        $method = $this->_getProtectedMethod($marshall, '_hasTranslations');
```

Comment:
Generally we try to avoid reflection hacks. Instead try focusing on testing the public interface and not internal implementation details.